keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/ec2/resource/ImageHandler.kt

```diff
@@ -0,0 +1,54 @@
+package com.netflix.spinnaker.keel.ec2.resource
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.module.kotlin.convertValue
+import com.netflix.spinnaker.keel.api.Resource
+import com.netflix.spinnaker.keel.api.ResourceKind
+import com.netflix.spinnaker.keel.api.ResourceName
+import com.netflix.spinnaker.keel.api.SPINNAKER_API_V1
+import com.netflix.spinnaker.keel.api.ec2.Image
+import com.netflix.spinnaker.keel.api.ec2.NamedImage
+import com.netflix.spinnaker.keel.clouddriver.CloudDriverService
+import com.netflix.spinnaker.keel.plugin.ResourceHandler
+import com.netflix.spinnaker.keel.plugin.ResourceNormalizer
+import de.danielbechler.diff.node.DiffNode
+import kotlinx.coroutines.runBlocking
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+
+class ImageHandler(
+  private val cloudDriverService: CloudDriverService,
+  override val objectMapper: ObjectMapper,
+  override val normalizers: List<ResourceNormalizer<*>>
+  ) : ResourceHandler<NamedImage> {
+
+  override val log: Logger by lazy { LoggerFactory.getLogger(javaClass) }
+  override val apiVersion = SPINNAKER_API_V1.subApi("ec2")
+  override val supportedKind = ResourceKind(
+    apiVersion.group,
+    "namedImage",
+    "namedImage"
+  ) to NamedImage::class.java
+
+  override fun generateName(spec: NamedImage) = ResourceName(
+    "ec2:image:${spec.account}:${spec.name}"
```

Comment:
Do we need region in here?

Reply:
the NamedImage is multiregion (amis is a map of region -> list of amiId)